src/main/java/bio/terra/cli/command/Main.java

```diff
@@ -61,18 +72,61 @@ public static void main(String... args) {
   public void run() {}
 
   /**
-   * Custom exception handler class that suppresses the stack trace when an exception is thrown.
-   * Instead, it just prints the exception message and exits the process.
+   * Custom handler class that intercepts all exceptions.
+   *
+   * <p>There are three categories of exceptions. All print a message to stderr and log the
+   * exception.
+   *
+   * <p>- User-facing = user can fix
```

Comment:
(Caveat: I may be dramatically over-thinking things here. Take with a grain of salt and feel free to ignore at your leisure!)

I thought I was grokking the categorization here until about 2/3 way through this PR, when I realized that both user-facing and Internal exceptions are getting shown to the user with very similar formatting... so now I'm having more trouble precisely distinguishing between the two use cases.

The main difference in handling seems to be that (1) for Internal exceptions we'll include a message from the causal exception, and (2) for Internal exceptions we point users to the log file for more details (presumably so they can find a stack trace).

But for both types of exceptions, we're showing users a big red '[ERROR]'-prefixed line with the exception message included. This feels quite similar to me, and may make it hard to describe clear conventions around what level of user-facing polish should be applied to each type of exception.

Spitballing a couple potential changes that may help draw out more of a difference:

1) Consider naming options.

Surveying the usage patterns, `UserFacingException` is mostly thrown when there is a simple precondition failure (e.g. "no workspace context") or a par-for-the-course error code returned from the service (e.g. "resource not found"). Maybe these could be called `UserActionableException` or simply `UserException`? Whereas `InternalErrorException` is usually about a server failure (5xx), file parsing failure, or some other internal workings of the system. Possibly `SystemException` captures that concisely? I realize this isn't too different from the current names... but somehow I found the `UserFacing` term slightly jarring, because ultimately all of these exceptions are shown to the user.

2) Consider a formatting difference between user-actionable and internal system exceptions.

If I use incorrect characters in an object name or try to fetch an ID that doesn't exist, I might expect a less-scary looking error message than if a server is down or there is an error parsing some core config file. To some extent, for the former type of "user error", I'd probably want it to look similar to what happens when I mis-type a command name. So maybe there's some stronger delineation we can aim for here in the formatted output, e.g.

UserActionable: normal color, less loud (maybe drop the [ERROR] prefix?), looks like a simple message "Hey, you did something wrong" or "hey, we couldn't find that thing you asked for".
Internal / system: some red somewhere (maybe just the [ERROR] prefix is red), with loud with clear boilerplate indicating that a relatively serious system error has occurred.

Reply:
This makes sense about there not being a big enough difference between user-actionable and system/unexpected exceptions.
1. Renamed the exception classes to UserActionableException and SystemException.
2. Changed the formatting so that system and unexpected exceptions show a bold red message, and user-actionable exceptions show a regular size blue message. Since command typing errors (e.g. forgot a required flag) are also user-actionable, I also changed them to follow this convention.

I also explicitly defined the color scheme in the `Main` class, so that we can modify it more easily in the future. Below is an example of what it looks like now (I inserted test exceptions in the `terra status` and `terra version` commands):
<img width="1270" alt="Screen Shot 2021-03-17 at 2 20 39 PM" src="https://user-images.githubusercontent.com/12446128/111518598-98650480-872c-11eb-92b3-1b595712ed5e.png">

Reply:
I don't have any strong preferences around the colors here, so if there's any changes you guys would like to see, just let me know. (There's no orange though FYI 🤷‍♀️ ).